misc/snapshotter/snapshotter.sh

```diff
@@ -41,9 +49,65 @@ function fs_driver_handler() {
     COMMANDLINE+=" --config ${SNAPSHOTTER_CONFIG}"
 }
 
+function configure_snapshotter() {
+
+    echo "configuring snapshotter"
+    if [ "${CONTAINER_RUNTIME}" != "containerd" ]; then
+        die "not supported container runtime: ${CONTAINER_RUNTIME}"
+    fi
+
+    # Copy the container runtime config to a backup
+    cp "$CONTAINER_RUNTIME_CONFIG" "$CONTAINER_RUNTIME_CONFIG".bak.nydus
+
+
+    # When trying to edit the config file that is mounted by docker with `sed -i`, the error would happend:
+    # sed: cannot rename /etc/containerd/config.tomlpmdkIP: Device or resource busy  
+    # The reason is that `sed`` with option `-i` creates new file, and then replaces the old file with the new one, 
+    # which definitely will change the file inode. But the file is mounted by docker, which means we are not allowed to 
+    # change its inode from within docker container.
+    # 
+    # So we copy the original file to a backup, make changes to the backup, and then overwrite the original file with the backup.
+    cp "$CONTAINER_RUNTIME_CONFIG" "$CONTAINER_RUNTIME_CONFIG".bak
+    # Check and add nydus proxy plugin in the config
+    if grep -q '\[proxy_plugins.nydus\]' "$CONTAINER_RUNTIME_CONFIG".bak; then
+        echo "the config has configured the nydus proxy plugin!"
+    else
+        echo "Not found nydus proxy plugin!"
+        cat <<EOF >>"$CONTAINER_RUNTIME_CONFIG".bak
+        
+[proxy_plugins]
+    [proxy_plugins.nydus]
+        type = "snapshot"
+        address = "$SNAPSHOTTER_GRPC_SOCKET"
+EOF
+    fi
+
+    if grep -q 'disable_snapshot_annotations' "$CONTAINER_RUNTIME_CONFIG".bak; then
```

Comment:
If the `disable_snapshot_annotations` line is commented:
```
    [plugins."io.containerd.grpc.v1.cri".containerd]
      default_runtime_name = "runc"
#      disable_snapshot_annotations = true
      discard_unpacked_layers = false
      ignore_rdt_not_enabled_errors = false
      no_pivot = false
      snapshotter = "overlayfs"
```

then setting its value to false won't take any effect anyway. Same rationale applies to other substitutions like `discard_unpacked_layers` below.

Edit toml files with cat/grep/sed is a s***! :(

I was about to suggest convert this script to Golang then use a higher-level library for that purpose. However, I glanced at the go-toml library, which seems used by containerd, but it doesn't provide a good in-place editing support (IIUC from the https://github.com/pelletier/go-toml/issues/457 issue). Does containerd provides an API that we could use to change its own configuration toml?